src/rgw/rgw_admin.cc

```diff
@@ -5531,6 +5531,48 @@ int main(int argc, const char **argv)
   } /* OPT_BUCKETS_LIST */
 
   if (opt_cmd == OPT_BUCKET_STATS) {
+    if (bucket_name.empty() && !bucket_id.empty()) {
+      void *handle;
+      int ret = store->ctl()->meta.mgr->list_keys_init("bucket.instance", marker, &handle);
+      if (ret < 0) {
+        cerr << "ERROR: can't get key: " << cpp_strerror(-ret) << std::endl;
+        return -ret;
+      }
+
+      bool truncated = false;
+      uint64_t count = 0;
+      string s;
+      string token;
+      string delimiter = ":";
+      uint64_t pos = 0;
+      bool bucket_name_from_bucket_id = false;
+      do {
+        list<string> keys;
+        ret = store->ctl()->meta.mgr->list_keys_next(handle, 1, keys, &truncated);
+        if (ret < 0) {
+          cerr << "ERROR: lists_keys_next(): " << cpp_strerror(-ret) << std::endl;
+          return -ret;
+        }
+        for (list<string>::iterator iter = keys.begin(); iter != keys.end(); ++iter) {
+           s = *iter;
+           while ((pos = s.find(delimiter)) != string::npos) {
```

Comment:
i don't quite follow this parsing loop. you seem to be splitting by : and assigning `token` to the second-to-last segment. why not the first segment?

there's a `rgw_bucket_parse_bucket_key()` in rgw_bucket.h/cc that's designed to parse these metadata keys, including the `tenant/bucket` part of the bucket name

```c++
// parse the tenant/name:instance from the metadata key
rgw_bucket bucket;
ret = rgw_bucket_parse_bucket_key(cct, s, &bucket, nullptr);
if (ret < 0) {
  continue;
}
if (bucket_id == bucket.bucket_id) {
  bucket_op.set_tenant(bucket.tenant);
  bucket_op.set_bucket_name(bucket.name);
  bucket_name_from_bucket_id = true;
  break;
}
```